lib/display_driver_utils.py

```diff
@@ -18,14 +19,15 @@ def __init__(self,width=420,height=320,orientation=ORIENT_PORTRAIT):
         self.disp = None
         self.touch = None
         self.type = None
-        if not lv_utils.event_loop.is_running():
+        if sys.platform == 'javascript' or not lv_utils.event_loop.is_running():
```

Comment:
Isn't that the other way around? If this is js port, we don't want to call `init_gui`.

Reply:
I think this is correct. I still use `init_gui` on the JS port to initialize SDL. The only difference is the event loop itself.

Reply:
I see, you want to prevent a call to `lv_utils.event_loop` because `lv_utils` wasn't imported...  

How about solving this differently: 

- Don't change anything in `display_driver_utils.py` or `lv_utils.py`
- Create a "dummy" Timer implementation for the JS port only.  
We could add a new JS specific `lv_timer.py` under `driver/js` and make sure `driver/js` is in the path (add it with `sys.path.append` [in the JS  port](https://github.com/lvgl/lv_micropython/blob/45c59a8f22395266c0079e115928adebe11bea2b/ports/javascript/frontend_src/main.js#L231)). It doesn't need to do anything really, just provide the Timer API.

Something like this:

```python
class Timer:

    PERIODIC = 0
    ONE_SHOT = 1

    def __init__(self, id):
        pass

    def init(self, mode=PERIODIC, period=-1, callback=None):
        pass

    def deinit(self):
        pass
```

With unchanged `display_driver_utils.py` this would create a "dummy" event loop that does nothing on the JS port, and also provide a boilerplate for a future Timer implementation for the JS port if we ever want that.  
It would also keep uniform `display_driver_utils.py` and `lv_utils.py` for all ports.

What do you think?

Reply:
I can do this, but I'd like it to be clear to someone tinkering with the internals that the timer is a stub and does nothing, otherwise they'll wonder why the event loop doesn't work. Maybe printing a warning on the second and future times the timer is initialized is good enough. The warning would be suppressed for the [first initialization in `lv_utils`](https://github.com/lvgl/lv_binding_micropython/blob/3974a78e8effe34bbcfc22293e43398cbc2d9027/lib/lv_utils.py#L84).

Reply:
> Maybe printing a warning on the second and future times the timer is initialized is good enough

Someone tinkering with the internals might not use the simulator directly. So he would probably initialize the display+event_loop only once and won't see any warning.

We can print the warning always but it would also be displayed on the simulator, which is a bit confusing.

I think that someone tinkering with the internals would sooner or later reach driver/js/lv_timer.py, so perhaps a clear comment there could be enough?

Reply:
I think so. I've done it this way in the latest commit. I'll merge this as it shouldn't affect anything else (just adds a new file under `driver/js`).